docs/backend_architecture/tasks.rst

```diff
@@ -94,12 +97,14 @@ A successful response looks like this:
     }
 
 When we send a request to ``POST /api/tasks/tasks/`` API endpoint, first, we validate the payload. The request
-payload **must** have a ``"task"`` parameter as string and the user should have the permissions mentioned on the
+payload **must** have a ``"type"`` parameter as string and the user should have the permissions mentioned on the
 ``permission_classes`` argument of decorator. If the user has permissions then we proceed.
 
-Then, we check whether the ``"task"`` function has a validator associated with it or not. If it has a validator, it
-gets run. The return value of the validator must be a dictionary. The dictionary returned by the validator is passed to the task function as keyword
-arguments.
+Then, we check whether the registered task function has a validator associated with it or not. If it has a validator, it
+gets run. The return value of the validator must be a dictionary that conforms to the function signature of the Job object.
+The dictionary returned by the validator is passed to a Job object to be enqueued. By default, any key value pairs in the
+request object that are registered as input fields on the validator will be passed to the function as kwargs. If no fields
+are defined on the validator, or no validator is registered, then the function will receive no arguments.
```

Comment:
So essentially this means every task function needs to decalre a validator for it to receive arguments? :thinking:

Reply:
Yes - the main motivation here is to ensure an explicit interface, so if you want to receive arguments, you have to define what those arguments will be using a validator, and because the validator uses serializer fields, you will also be specifying the types as well.

Reply:
Agreed! Explicity is always better!